### PR TITLE
feat: refactor favorite items

### PR DESCRIPTION
**Pull request by morganridel**

### CODE REVIEW

File: cypress/support/server.ts [hunk 1]
```diff
@@ -1940,3 +1942,59 @@ export const mockUpdatePassword = (
...
+};
+
+export const mockAddFavorite = (
+  _items: ItemForTest[],
+  shouldThrowError: boolean,
+): void => {
+  cy.intercept(
+    {
+      method: HttpMethod.POST,
+      url: new RegExp(`${API_HOST}/${GET_FAVORITE_ITEMS_ROUTE}/${ID_FORMAT}`),
```
Comment (pyphilia) [inline cypress/support/server.ts:1972]: It doesn't have to be changes, but if possible you can use the API endpoints from query-client, so it doesn't break if we change the endpoint route.

File: package.json [hunk 2]
```diff
@@ -147,7 +147,8 @@
     "nth-check": "2.1.1",
     "react-error-overlay": "6.0.11",
     "@types/react": "17.0.62",
-    "@svgr/webpack": "6.5.1"
+    "@svgr/webpack": "6.5.1",
+    "@graasp/sdk": "github:graasp/graasp-sdk#refactor-favorite"
```
Comment (pyphilia) [inline package.json:151]: Don't forget to change your branches ;)

File: src/components/common/FavoriteButton.tsx [hunk 3]
```diff
@@ -19,48 +21,35 @@ type Props = {
...
 };
 
-export const isItemFavorite = (
+const isItemFavorite = (
   item: ItemRecord,
-  member?: MemberRecord,
-): boolean => member?.extra?.favoriteItems?.includes(item.id) || false;
-
+  favorites?: List<ItemFavoriteRecord>,
+): boolean => favorites?.map((f) => f.item.id).contains(item.id) || false;
```
Comment (pyphilia) [inline src/components/common/FavoriteButton.tsx:27]: Why not use `.some()`?

**Review by pyphilia — CHANGES_REQUESTED** [review]
LGTM! I've left minor comments ⭐

File: cypress/support/server.ts (hunk 1 above)
Comment (morganridel) [reply cypress/support/server.ts:1972]: The problem is that the endpoint builder takes a UUID to build the url, and the ID_FORMAT is a string

File: src/components/common/FavoriteButton.tsx (hunk 3 above)
Comment (morganridel) [reply src/components/common/FavoriteButton.tsx]: I didn't know about it :)

**Pull request merged**